cpp/src/arrow/compute/kernels/scalar_nested.cc

```diff
@@ -196,9 +196,22 @@ const FunctionDoc list_element_doc(
 struct StructFieldFunctor {
   static Status Exec(KernelContext* ctx, const ExecSpan& batch, ExecResult* out) {
     const auto& options = OptionsWrapper<StructFieldOptions>::Get(ctx);
-
     std::shared_ptr<Array> current = MakeArray(batch[0].array.ToArrayData());
-    for (const auto& index : options.indices) {
+
+    // Specific struct handling
```

Comment:
I don't understand why this is special-cased. I would intuitively expect the following process:
1) compute a `FieldPath` using `FieldRef::FindOne`
2) Loop over the indices of the `FieldPath` in the loop below

Reply:
Indeed. Think I was just on a bender for dealing with structs. This way works and is easier on the eyes.